tests/features/syntax_highlight_test.py

```diff
@@ -160,3 +160,20 @@ def test_syntax_highlighting_tabs_after_line_creation(run, tmpdir):
 def test_does_not_crash_with_no_color_support(run):
     with run(term='xterm-mono') as h, and_exit(h):
         pass
+
+
+def test_file_without_extension(run, tmpdir):
+    f = tmpdir.join('f')
+    f.write('#!/usr/bin/env demo\n' + DEMO_S)
+    with run(str(f), term='screen-256color', width=20) as h, and_exit(h):
+        for i, attr in enumerate([
+                [(236, 40, curses.A_REVERSE)] * 20,        # header
+                [(243, 40, 0)] * 19 + [(236, 40, 0)],      # shebang
+                [(52, 203, 0)] * 5 + [(236, 40, 0)] * 15,  # - foo
+                [(243, 40, 0)] * 14 + [(236, 40, 0)] * 6,  # # comment here
+                [(236, 40, 0)] * 20,                       # uncolored
+                [(17, 40, 0)] * 7 + [(236, 40, 0)] * 13,   # """tqs!
+                [(17, 40, 0)] * 10 + [(236, 40, 0)] * 10,  # still more
+                [(17, 40, 0)] * 3 + [(236, 40, 0)] * 17,   # """
+        ]):
+            h.assert_screen_attr_equals(i, attr)
```

Comment:
this test doesn't fail without the change -- this is demoing "firstLineMatch" and not the identify code -- you should be able to write a test similar to the original test